### PR TITLE
trace/capabilities: Implement unit tests

### PR DESCRIPTION
**Pull request by mauriciovasquezbernal**
Fixes https://github.com/kinvolk/inspektor-gadget/issues/830

### CODE REVIEW

File: pkg/gadgets/trace/capabilities/tracer/tracer_test.go [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+					Comm:      info.Comm,
+					CapName:   "CAP_CHOWN",
+					Cap:       0,
+					Audit:     1,
+					InsetID:   &false_,
+					Verdict:   "Allow",
+				}
+			}),
+		},
+		"CAP_CHOWN": {
```
Comment (alban) [inline pkg/gadgets/trace/capabilities/tracer/tracer_test.go:146]: Is it a duplicate from the `captures_events_with_matching_filter` test above?

**Review by alban — COMMENTED** [review]
I like how the tests are getting comprehensive. Some comments.

File: pkg/gadgets/internal/test/helpers.go [hunk 2]
```diff
@@ -64,6 +65,17 @@ func RequireRoot(t *testing.T) {
 	}
 }
 
+func RequireKernelVersion(t *testing.T, expectedVersion *kernel.VersionInfo) {
+	version, err := kernel.GetKernelVersion()
```
[the remark this answers is not in the document]
Comment (mauriciovasquezbernal) [reply pkg/gadgets/internal/test/helpers.go:69]: `kernel.GetKernelVersion()` is also used in `pkg/gadgets/profile/block-io/tracer/tracer.go`. I checked it very quickly and it seems to me they're different:
- `github.com/moby/moby/pkg/parsers/kernel`: Provides a way to get and compare kernel versions.
- `github.com/cilium/ebpf/features`: Only provides a way to get the `LINUX_VERSION_CODE`. 

I agree that we could do some manual work to only use one of the libraries, but in the mean time I think we can keep both.

File: pkg/gadgets/trace/capabilities/tracer/tracer_test.go [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("running command: %w", err)
+	}
+
+	return nil
+}
+
+// chown requires CAP_CHOWN
+func chown() error {
+	file, err := ioutil.TempFile("/tmp", "prefix")
```
[the remark this answers is not in the document]
Comment (mauriciovasquezbernal) [reply pkg/gadgets/trace/capabilities/tracer/tracer_test.go:378]: Good catch. I'm wondering why the linter is not complaining about it.

File: pkg/gadgets/trace/capabilities/tracer/tracer_test.go [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+					Audit:     1,
+					InsetID:   &false_,
+					Verdict:   "Allow",
+				}
+			}),
+		},
+		"verdict_deny": {
+			getTracerConfig: func(info *utilstest.RunnerInfo) *tracer.Config {
+				return &tracer.Config{
+					MountnsMap: utilstest.CreateMntNsFilterMap(t, info.MountNsID),
```
[the remark this answers is not in the document]
Comment (mauriciovasquezbernal) [reply pkg/gadgets/trace/capabilities/tracer/tracer_test.go:160]: Good point. I fixed by adding `t.Cleanup(func() { m.Close() })` in `CreateMntNsFilterMap`. 

Regarding the fd leaks, I think it's possible to use it, but I'd like to find a way to make it as less verbose as possible. What about thinking about it in a future PR?

File: pkg/gadgets/trace/capabilities/tracer/tracer_test.go (hunk 1 above)
Comment (mauriciovasquezbernal) [reply pkg/gadgets/trace/capabilities/tracer/tracer_test.go]: Yes, you're right. I removed it and rename other cases.

File: pkg/gadgets/trace/capabilities/tracer/tracer_test.go [hunk 5]
```diff
@@ -0,0 +1,421 @@
...
+
+	return tracer
+}
+
+func generateNonAudit() error {
+	if err := chown(); err != nil {
+		return fmt.Errorf("chown(): %w", err)
+	}
+
+	// This command will generate some non-audit capabilities checks
```
[the remark this answers is not in the document]
Comment (mauriciovasquezbernal) [reply pkg/gadgets/trace/capabilities/tracer/tracer_test.go:326]: > Does it always? I thought some kernels might not.

TBD I'm not sure. It's a way I found to generate non-audit checks. Are you aware of another mechanism to generate it? 

> Why run both chown and cat in this function?

I wanted to generate different capabilities to check it was filtering out the non-audit ones. But I agree running only cat in this case should be enough.

**Review by mauriciovasquezbernal — COMMENTED** [review]
I think I handled most of your feedback. Thanks!

File: pkg/gadgets/trace/capabilities/tracer/tracer_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (alban) [reply pkg/gadgets/trace/capabilities/tracer/tracer_test.go:326]: > TBD I'm not sure. It's a way I found to generate non-audit checks. Are you aware of another mechanism to generate it?

No, I don't know.

**Comment by mauriciovasquezbernal** [comment]
@alban is there anything else I should do to merge it?

**Review by alban — APPROVED** [review]
LGTM

Sorry for the delay in reviewing.

**Comment by mauriciovasquezbernal** [comment]
Thanks for the review.

**Pull request merged**